app/models/papi_client.rb

```diff
@@ -9,18 +9,30 @@
 class PapiClient < Struct.new(:project, :service_account_credentials, :service)
   extend ServiceAccountManager
 
-   # Google authentication scopes necessary for running pipelines
+  # Google authentication scopes necessary for running pipelines
   GOOGLE_SCOPES = %w(https://www.googleapis.com/auth/cloud-platform)
+
   # Network and sub-network names, if needed
   GCP_NETWORK_NAME = ENV['GCP_NETWORK_NAME']
   GCP_SUB_NETWORK_NAME = ENV['GCP_SUB_NETWORK_NAME']
-  # List of scp-ingest-pipeline actions and their allowed file types
+
+   # List of scp-ingest-pipeline actions and their allowed file types
   FILE_TYPES_BY_ACTION = {
-      ingest_expression: ['Expression Matrix', 'MM Coordinate Matrix'],
-      ingest_cluster: ['Cluster'],
-      ingest_cell_metadata: ['Metadata'],
-      ingest_subsample: ['Cluster']
-  }
+    ingest_expression: ['Expression Matrix', 'MM Coordinate Matrix'],
+    ingest_cluster: ['Cluster'],
+    ingest_cell_metadata: ['Metadata'],
+    ingest_subsample: ['Cluster'],
+    differential_expression: ['Cluster']
+  }.freeze
+
+  # list of parameter names allowed for differential_expression calls
+  DE_PARAMETER_NAMES = %w[
+    annotation_name annotation_type annotation_scope annotation_file
+    cluster_file cluster_name matrix_file_path matrix_file_type
+  ].freeze
+
```

Comment:
I think this might be a case where a small class (possibly embeddable if we ever need to store it along with an ingest_job), akin to expression_file_info.rb, to store these parameters will really help clarify them, since then it's easier to inline descriptions of what each parameter does.  Likewise, then instead of having a generic `extra_options` parameter which requires some code-sleuthing to figure out what it accepts, the `get_command_line` function can just take an optional `differential_expression_params` object.

Reply:
Yeah ok I'll bite on that - I can see a non-persisted `ActiveModel` based class for this, akin to `FireCloudProfile` (something that only lives in memory but still acts like a model).

Reply:
[8f7fef6](https://github.com/broadinstitute/single_cell_portal_core/pull/1483/commits/8f7fef66d1357c74e4dea90ce08a41f8fb81ae98)